libpromises/locks.c

```diff
@@ -177,29 +180,12 @@ static inline void log_lock(const char *op,
     }
 }
 
-static void GenerateMd5Hash(const char *istring, char *ohash)
+static void HashLockKeyIfNecessary(const char *const istring, char *const ohash)
 {
-    if (!strcmp(istring, "CF_CRITICAL_SECTION"))
-    {
-        strcpy(ohash, istring);
-        return;
-    }
-
-    unsigned char digest[EVP_MAX_MD_SIZE + 1];
-    HashString(istring, strlen(istring), digest, HASH_METHOD_MD5);
-
-    const char lookup[]="0123456789abcdef";
-    for (int i=0; i<16; i++)
-    {
-        ohash[i*2]   = lookup[digest[i] >> 4];
-        ohash[i*2+1] = lookup[digest[i] & 0xf];
-    }
-    ohash[16*2] = '\0';
-
-    if (!strncmp(istring, "lock.track_license_bundle.track_license", 39))
-    {
-        ohash[0] = 'X';
-    }
+    assert(LMDB_MAX_KEY_SIZE == 511);
```

Comment:
This *assert* doesn't make much sense here. The function should work with any value for `LMDB_MAX_KEY_SIZE` that passes the following asserts, right?

Reply:
It's just an attempt to get people to review this code if they ever change the constant.

Reply:
Removed.